src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,209 @@
+          ResultSet rs = pst.executeQuery()) {
+        // Defines the url to which the user will be redirected. Will be defined each case because
+        // URL will change once notifications for approvals and meetings are enabled.
+        String elementUrl = "/questions.html?id=" + modifiedElementId;
+        // Insert notification and get its id to relate in UserNotification table.
+        insertToNotification(connection, "Somebody commented your answer", elementUrl, 
+                             localTimestamp);
+        int notificationId = getNotificationId(connection, localTimestamp);
+        // Iterate through the query's result set to insert all notifications.
+        while (rs.next()) {
+          insertToUserNotification(connection, rs.getInt(1), notificationId);
+        }
+        // Close the connection once all insertions have been performed.
+        connection.close();
+      } catch (SQLException ex) {
+          Logger lgr = Logger.getLogger(DataServlet.class.getName());
+          lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    }
+  }
+
+  /**
+  * Fetches notifications with the user id.
+  */
+  private List<Notification> getNotifications(int userId) {
+    // Prepares query to select notifications by the subquery of notifications IDs selected by 
+    // user ID.
+    String query =  "SELECT message, url, date_time FROM Notification WHERE id IN " +
+                    "(SELECT notification_id FROM UserNotification WHERE user_id = " + userId +
+                    ") ORDER BY date_time DESC";
+    List<Notification> notifications = new ArrayList<>();
+    // Query the information from tables and create notification object to be stored in ArrayList.
+    try (Connection connection = DriverManager.getConnection(Utility.SQL_LOCAL_URL,
+                                                             Utility.SQL_LOCAL_USER,
+                                                             Utility.SQL_LOCAL_PASSWORD);
+        PreparedStatement pst = connection.prepareStatement(query);
+        ResultSet rs = pst.executeQuery()) {
+      // Iterate through the result of the query to populate the ArrayList and return it as JSON.
+      while(rs.next()){
+        Notification notification = new Notification();
+        notification.message = rs.getString(1);
+        notification.url = rs.getString(2);
+        notification.timestamp = rs.getTimestamp(3);
+        // Stores object in ArrayList.
+        notifications.add(notification);
+      }
+      connection.close();
+    } catch (SQLException ex) {
+        Logger lgr = Logger.getLogger(DataServlet.class.getName());
+        lgr.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+    return notifications;
+  }
+
+  /**
+  * Function that receives the data for a notification and inserts it into the Notification table.
+  */
+  private void insertToNotification(Connection connection, String message, String elementUrl,
```

Comment:
Delete `connection` variable if you are not using it inside the function.

Reply:
It is being used.

Reply:
👍